portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +28,25 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Ryan!</h1>");
+
+    //create and populate animals ArrayList
+    ArrayList<String> animals = new ArrayList {"Bear", "Pig", "Goat"};
+
+    //Convert animals ArrayList to JSON
+    String json = convertToJsonUsingGson(animals);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts an ArrayList of Strings into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(ArrayList<String> strings) {
```

Comment:
Optional/nit: consider hiding implementation details like the expectation of ArrayList. You could use List or even go more generic, because you don't really care that the caller of this function is providing an ArrayList.

I realized that this is from serverstats example, and this is not too important to focus on, but I commented it on Angela's PR so I figured I'd do it here as well.

(also I'd just return gson.Json rather than storing it in a variable and then returning).